connectors/confluence/connector/src/main/java/org/apache/manifoldcf/crawler/connectors/confluence/client/ConfluenceClient.java

```diff
@@ -0,0 +1,717 @@
+
+			String url = String.format("%s://%s:%s/%s/%s?limit=1", protocol, host,
+					port, path, CONTENT_PATH);
+			logger.debug(
+					"[Processing] Hitting url: {} for confluence status check fetching : ",
+					"Confluence URL", sanitizeUrl(url));
+			HttpGet httpGet = createGetRequest(url);
+			response = httpClient.execute(httpGet);
+			int statusCode = response.getStatusLine().getStatusCode();
+			if (statusCode != 200)
+				throw new Exception(
+						"[Checking connection] Confluence server appears to be down");
+			else
+				return true;
+		} catch (IOException e) {
+			logger.warn(
+					"[Checking connection] Confluence server appears to be down",
+					e);
+			throw new Exception("Confluence appears to be down", e);
+		}
+	}
+
+	/**
+	 * <p>Check method used to test if Confluence instance is up and running when using Authority connector (JSON-RPC API)</p>
+	 * <p>This method will be deleted when all JSON-RPC methods are available through the REST API
+	 * 
+	 * @return a {@code Boolean} indicating whether the Confluence instance is alive or not
+	 * 
+	 * @throws Exception
+	 */
+	public boolean checkAuth() throws Exception {
+		try {
+			if (httpClient == null) {
+				connect();
+			}
+			getSpaces();
+			return true;
+		} catch (Exception e) {
+			logger.warn(
+					"[Checking connection] Confluence server appears to be down",
+					e);
+			throw e;
+		}
+	}
+	/**
+	 * <p>
+	 * Create a get request for the given url
+	 * </p>
+	 * 
+	 * @param url
+	 *            the url
+	 * @return the created {@code HttpGet} instance
+	 */
+	private HttpGet createGetRequest(String url) {
+		String finalUrl = useBasicAuthentication() ? url + "&os_authType=basic": url;
+		String sanitizedUrl = sanitizeUrl(finalUrl);
+		HttpGet httpGet = new HttpGet(sanitizedUrl);
+		httpGet.addHeader("Accept", "application/json");
+		if (useBasicAuthentication()) {
+			httpGet.addHeader(
+					"Authorization",
+					"Basic "
+							+ Base64.encodeBase64String(String.format("%s:%s",
+									this.username, this.password).getBytes(
+									Charset.forName("UTF-8"))));
+		}
+		return httpGet;
+	}
+
+	/**
+	 * <p>
+	 * Get a list of Confluence pages
+	 * </p>
+	 * 
+	 * @return a {@code ConfluenceResponse} containing the result pages and
+	 *         some pagination values
+	 * @throws Exception
+	 */
+	public ConfluenceResponse<Page> getPages() throws Exception {
+		return getPages(0, 50, Optional.<String> absent());
+	}
+
+	/**
+	 * <p>
+	 * Get a list of Confluence pages using pagination
+	 * </p>
+	 * 
+	 * @param start The start value to get pages from
+	 * @param limit The number of pages to get from start
+	 * @return a {@code ConfluenceResponse} containing the result pages and
+	 *         some pagination values
+	 * @throws Exception
+	 */
+	@SuppressWarnings("unchecked")
+	public ConfluenceResponse<Page> getPages(int start, int limit,
+			Optional<String> space) throws Exception {
+		String url = String.format("%s://%s:%s/%s/%s?limit=%s&start=%s", protocol,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code uses &#39;%s&#39; to format integers: start (declared at line 267), limit (declared at line 267). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 281. Consider formatting this integer with &#39;%d&#39; instead.